### PR TITLE
Send and Receive Requests from Other Users to Connect

### PR DESCRIPTION
**Pull request by safchow**
## Trello ticket link

[As a user, I want to send requests to other users for their contact information so that I can connect externally with them.](https://trello.com/c/vSjN7pkB/12-as-a-user-i-want-to-send-requests-to-other-users-for-their-contact-information-so-that-i-can-connect-externally-with-them)
[As a user, I want to see requests I've received from other users so that I can connect externally with them.](https://trello.com/c/kauXCwth/126-as-a-user-i-want-to-see-requests-ive-received-from-other-users-so-that-i-can-connect-externally-with-them)
[As a user, I want to see requests I've sent to other users.](https://trello.com/c/6JIOJyul/130-as-a-user-i-want-to-see-requests-ive-sent-to-other-users)

## Implementation description
* Users can send requests (short messages) to other users to connect on their profiles.
* Sent and received requests can be seen on the requests page (`/requests`)
* Requests page can be accessed through the nav bar (by clicking the email icon)

## Steps to test
1. Send a request to another user (not yourself)
2. On the requests page, verify that it appears as a sent request
3. Go to `Requests.js`, and replace `parsed[0].id` on lines 35 and 36 to the id of the user you sent a request to. Ex:
    ```
    handleGetRequestSent(parsed[0].id);
    handleGetRequestReceived(parsed[0].id);
    // to
    handleGetRequestSent(3);
    handleGetRequestReceived(3);
    ```
4. Verify that the request you sent to that user appears as a request received from you (you're essentially viewing the page as them, verify that behaviour is correct)
5. Back to sending requests: verify that an empty body is not allowed
6. Verify that a successful toast message appears after sending a valid request
7. NEW STEP: verify that you cannot send a user more than one request, verify that you can still send other users requests (as long as its the first one)

## What should reviewers focus on?
* Are there any styling issues? Are form elements clear and concise?

### CODE REVIEW

**Review by c45choi — CHANGES_REQUESTED** [review]
only thing is does it make sense to send multiple requests to the same person?

also maybe f this but how about deleting a request...

**Comment by c45choi** [comment]
this is fire tho im so impressed

**Comment by safchow** [comment]
> only thing is does it make sense to send multiple requests to the same person?
> 
> also maybe f this but how about deleting a request...

This would be its own user story and has some complications:
- If a user who received a request deletes it, then the user who sent the request has the request deleted on their end as well. It isn't very easy to make it so that one user sees the request they sent, and the other does not see the request they received (since they chose to ignore it or something). 
- There has to be a request_status column or something to store the response of the user who received the request
- If the value in the column is something like "ignore" then don't render the message on the received user side

The idea is that the user sends a request, and they connect outside of the platform. Sending multiple requests can be an issue, but since we're implementing this at the end of the sprint there isn't much time to fix the sections that are a bit iffy

**Review by c45choi — APPROVED** [review]
your points are valid, i think in our demo and stuff lets just make it seem like we can request once haha!

**Comment by safchow** [comment]
> your points are valid, i think in our demo and stuff lets just make it seem like we can request once haha!

I can add an error catch to ensure the user isn't sending another request to the same user, should be a quick fix.

**Comment by c45choi** [comment]
> > your points are valid, i think in our demo and stuff lets just make it seem like we can request once haha!
> 
> I can add an error catch to ensure the user isn't sending another request to the same user, should be a quick fix.

im fine with anything its not super high priority

File: src/components/users/Requests.js
Comment (gabbytang) [inline src/components/users/Requests.js:1]: There's an error in the Requests page. The Received requests list my name as the person who sent me the request. For example, 
image

I sent the bottom request from "Another Test Account" to my Gabrielle Tang account. In the received request box, it says I sent myself the message.

**Review by gabbytang — CHANGES_REQUESTED** [review]
Otherwise the issue I mentioned, the functionality works great!

**Comment by safchow** [comment]
> > > your points are valid, i think in our demo and stuff lets just make it seem like we can request once haha!
> > 
> > 
> > I can add an error catch to ensure the user isn't sending another request to the same user, should be a quick fix.
> 
> im fine with anything its not super high priority

added error catch for this

File: src/components/users/Requests.js
Comment (safchow) [reply src/components/users/Requests.js:1]: Added a fix for this, lmk if it works

**Review by gabbytang — APPROVED** [review]
Issue has been fixed and test step 7 is successful as well

**Pull request merged**